### PR TITLE
Manage libnginx-mod-http-passenger on modern Debian / Ubuntu

### LINKED ISSUE #1340: libnginx-mod-http-passenger should be installed instead of passenger
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: Any
- Ruby: Any
- Distribution: Debian 9 and Ubuntu 18.04+
- Module version: Any

## How to reproduce (e.g Puppet code you use)

add the following to hiera on a debian 9 or Ubuntu 18.04 and apply then nginx module
```
nginx::package_source: passenger
nginx::package_name: nginx-extras
nginx::http_cfg_append:
  'passenger_root': '/usr/lib/ruby/vendor_ruby/phusion_passenger/locations.ini'
  'passenger_ruby': '/usr/bin/passenger_free_ruby
```
## What are you seeing

Nginx throws an error regarding passenger_root
```
nginx: configuration file /etc/nginx/nginx.conf test failed
nginx: [emerg] unknown directive "passenger_root" in /etc/nginx/nginx.conf:64
```

## What behaviour did you expect instead

A valid nginx configuration thats able to understand passenger.

## Any additional information you'd like to impart

The passenger install guide says to use libnginx-mod-http-passenger and not passenger for Debian 9 and Ubuntu 18.04. libnginx-mod-http-passenger will pull passenger as a dependency.

https://www.phusionpassenger.com/docs/tutorials/deploy_to_production/installations/oss/ownserver/ruby/nginx/

### PR DESCRIPTION
**Pull request by smortex**
#### Pull Request (PR) description
This is a follow-up to #1341 (closed without being merged) which was an attempt to fix #1340. I took the commits in this legacy PR, rebased them on top of master (fixing a bunch of conflicts) and am now fixing the CI.

#### This Pull Request (PR) fixes the following issues
Fixes #1340

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -64,6 +64,7 @@
   Boolean $super_user                                        = $nginx::params::super_user,
   $temp_dir                                                  = $nginx::params::temp_dir,
   Boolean $server_purge                                      = false,
+  Boolean $include_modules_enabled                           = $nginx::params::include_modules_enabled,
```
Comment (bastelfreak) [inline manifests/init.pp]: can you please add documentation for the new parameters?

File: manifests/init.pp [hunk 2]
```diff
@@ -166,6 +171,7 @@
   Boolean $mime_types_preserve_defaults                      = false,
   Optional[String] $repo_release                             = undef,
   $passenger_package_ensure                                  = 'present',
+  String $passenger_package_name                             = $nginx::params::passenger_package_name,
```
Comment (bastelfreak) [inline manifests/init.pp]: Same here. Please document the parameter and enforce a minimal string length of 1:
```suggestion
  String[1] $passenger_package_name                          = $nginx::params::passenger_package_name,
```

Comment (smortex) [reply manifests/init.pp]: Sorry for the delay, I just pushed a fix!

**Comment by bastelfreak** [comment]
I'm not sure if this is a backwards-incompatible release or not. What about those newer Debian versions if they currently try to use the module? Doesn't it work at all? What happens if the module installed the passanger package in the past and now install the new package? Does that work at all?

**Comment by smortex** [comment]
Back from holidays, sorry for the delay!

The `libnginx-mod-http-passenger` package depends on the `passenger` package which was previously including everything.  The new package installs the actual nginx shlib and the adequate configuration file.

Per se, the previous behavior is working from the Puppet point of view, but the resulting configuration is not working (because the nginx module is not installed nor configured).

My guess is that this would only require a MINOR version bump: the change is backward compatible but adds support for systems previously unsupported.

Thanks!

**Pull request merged**